app/connectors/check/CheckConnector.scala

```diff
@@ -0,0 +1,113 @@
+  protected lazy val serviceUrl: String = servicesConfig.baseUrl("business-rates-check")
+
+  private def url(path: String) = s"$serviceUrl/$path"
+
+  def start(
+        propertyLinkId: PropertyLinkId,
+        assessmentRef: AssessmentRef,
+        checkType: CheckType,
+        propertyLinkSubmissionId: Option[String] = None,
+        uarn: Option[Long] = None,
+        dvrCheck: Boolean = false,
+        rateableValueTooHigh: Option[Boolean] = None
+  )(implicit hc: HeaderCarrier): Future[Either[RequestFailure, CheckId]] = {
+    implicit val httpReads: HttpReads[Either[RequestFailure, HttpResponse]] =
+      createReads(CREATED, Map(FORBIDDEN -> Forbidden))
+    http
+      .POSTEmpty[Either[RequestFailure, HttpResponse]](
+        url(
+          s"property-link/$propertyLinkId/assessment/$assessmentRef/start-check/${checkType.value}${queryParams(propertyLinkSubmissionId, uarn, Some(dvrCheck), rateableValueTooHigh)}"
+        )
+      )
+      .map {
+        case Right(response) =>
+          Right(response.json.as[CheckIdWrapper].id)
+        case Left(failure) => Left(failure)
+      }
+  }
+
+  def updateResumeCheckUrl(checkId: CheckId, resumeUrl: String)(
+        implicit hc: HeaderCarrier
+  ): Future[Either[RequestFailure, RequestSuccess]] = {
+
+    implicit val httpReads = createReads(NO_CONTENT, Map(FORBIDDEN -> Forbidden, NOT_FOUND -> NotFound))
+
+    http.PUT[JsValue, Either[RequestFailure, HttpResponse]](
+      url(s"partial-check/$checkId/resume"),
+      Json.obj("resumeUrl" -> resumeUrl)
+    ) map {
+      case Right(_)      => Right(Success)
+      case Left(failure) => Left(failure)
+    }
+  }
+
+  private def createReads(okStatus: Int, errorMap: Map[Int, RequestFailure]) =
+    new HttpReads[Either[RequestFailure, HttpResponse]] {
+
+      def read(method: String, url: String, response: HttpResponse): Either[RequestFailure, HttpResponse] =
+        response.status match {
+          case status if status == okStatus        => Right(response)
+          case status if errorMap.contains(status) => Left(errorMap(status))
+          case unexpectedStatus =>
+            logger.warn(s"[$method $url] failed: status: $unexpectedStatus, body: ${response.body}")
+            throw ConnectorException(method, url, response)
+        }
+    }
+
+  private def queryParams(
+        propertyLinkSubmissionId: Option[String],
+        uarn: Option[Long],
+        dvrCheck: Option[Boolean] = None,
+        rateableValueTooHigh: Option[Boolean] = None
+  ): String =
+    Seq(
+      propertyLinkSubmissionId.map(p => s"propertyLinkSubmissionId=$p"),
+      uarn.map(u => s"uarn=$u"),
+      dvrCheck.map(d => s"dvrCheck=$d"),
+      rateableValueTooHigh.map(r => s"rateableValueTooHigh=$r")
+    ).flatten.mkString("?", "&", "")
+
+  private case class CheckIdWrapper(id: CheckId)
+
+  private object CheckIdWrapper extends PartialCheckFormats {
+    implicit val format: Reads[CheckIdWrapper] = Json.reads[CheckIdWrapper]
+  }
+
```

Comment:
Happy to be told otherwise but this could potentially be refactored to make use of the BusinessRatesCheckService layer & possibly move CheckIdWrapper into its own model?